src/txorphanage.cpp

```diff
@@ -74,21 +99,43 @@ int TxOrphanage::_EraseTx(const uint256& txid)
             m_outpoint_to_orphan_it.erase(itPrev);
     }
 
-    size_t old_pos = it->second.list_pos;
-    assert(m_orphan_list[old_pos] == it);
-    if (old_pos + 1 != m_orphan_list.size()) {
-        // Unless we're deleting the last entry in m_orphan_list, move the last
-        // entry to the position we're deleting.
-        auto it_last = m_orphan_list.back();
-        m_orphan_list[old_pos] = it_last;
-        it_last->second.list_pos = old_pos;
+    if (it->second.list_pos >= 0) {
```

Comment:
Should this function ever be called with a protected orphan? Tests don't cover these lines for protected peers(added an assert, never hit), and seems suspect that it would delete things below this block if it's protected.

Reply:
> Should this function ever be called with a protected orphan?

Yes. It's called when orphans expire as well.

> Tests don't cover these lines for protected peers

Sounds like I neglected to write a test for this. Expiry is pretty long, but theoretically we can hit this if we have multiple peers stalling us on the resolution of this orphan. We might happen to be trying package relay with one of the peers when the expiration is reached.

Reply:
Ah! Didn't put 2 and 2 together. Comment-worthy